packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/converter/convertDiagram.ts

```diff
@@ -32,6 +32,30 @@ import { RectangularNodeData } from '../renderer/node/RectangularNode.types';
 
 const defaultPosition: XYPosition = { x: 0, y: 0 };
 
+/**
+ * Infers the primary, vertical and horizontal alignments from the inside label location.
+ *
+ * @param insideLabel
+ * @returns [primary alignment, vertical alignment, horizontal alignment]
+ */
+const getAlignment = (insideLabelLocation: InsideLabelLocation): [string, string, string] | [] => {
```

Comment:
In this version of the code, `InsideLabelLocation` is defined to be 'TOP_CENTER' and nothing else. I understand that we'll need a more complex mapping, but this could be `return ['TOP', 'TOP', 'CENTER'];` (I think) in this version.
I'm not even sure that's the correct return value because the logic is not that easy to understand.

Even when we support all variants, I'm wondering if a simple map lookup would not be enough, and probably more readable and maintainable?

```js
const ALIGNMENT_MAP = {
  'TOP_CENTER': ['TOP', 'TOP', 'CENTER'],
  // the others added later
}
const const getAlignment = (insideLabelLocation: InsideLabelLocation): [string, string, string] | [] => {
  return ALIGNMENT_MAP[insideLabelLocation] || [];
}
```

Reply:
Re-reading this, the use of an array of 3 strings should probaly be replace by object type with properly named fields.

Reply:
Done